src/xbot/common/controls/sensors/Gamepad.java

```diff
@@ -1,52 +1,109 @@
 package xbot.common.controls.sensors;
 
+import edu.wpi.first.wpilibj.GamepadBase;
 import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
 import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
 import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter;
 import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter.GamepadComponent;
+import xbot.common.math.XYPair;
 
 public class Gamepad implements XGamepad {
     
-    final GenericHID internalHID;
-    
-    final XJoystick leftStick;
-    final XJoystick rightStick;
-    final XJoystick dpad;
-    final XJoystick leftTrigger;
-    final XJoystick rightTrigger;
+    final XboxController internalGamepad;
     
     public Gamepad(int port) {
-        internalHID = new Joystick(port);
         
-        leftStick = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.LeftJoystick);
-        rightStick = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.RightJoystick);
-        dpad = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.DPad);
-        leftTrigger = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.LeftTrigger);
-        rightTrigger = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.RightTrigger);
-    }
-
-    @Override
-    public XJoystick getLeftStick() {
-        return leftStick;
+        internalGamepad = new XboxController(port);
     }
 
     @Override
-    public XJoystick getRightStick() {
-        return rightStick;
-    }
+    //double
+    //Joysticks
```

Comment:
I like this comment; can you make it more obvious? Maybe something like this (use an intentional number of dashes):

```java
// Joysticks --------------------------------------------------------------------
```